### ISSUE TITLE
Request: core info should check the hash of bios files when a hash is available

### ISSUE THREAD

**Issue by kcin2001** [libretro/RetroArch#7917]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

To make verifying bios files easier for the user it should hash the bios file to verify that it is the correct bios file

### Expected behavior

bios file hashing

### Actual behavior

just name checking

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]

### Environment information

- OS: [The operating system you're running]
- Compiler: [In case you are running local builds]

**Comment by orbea** [comment]
With some cores the user's working bios files will not match the hashes for working bios files which other users have.

**Comment by bparker06** [comment]
The fact that we would have to scrape the freeform description field is also a problem, unless the core info format was modified.

**Comment by i30817** [comment]
You mean it doesn't already? Filename check only? Uh.

**Comment by retro-wertz** [comment]
why stop there? i think its better after checksum the files, especially those not marked option can be loaded ahead and passed to core (similarly to loading content/rom),

**Comment by bparker06** [comment]
>Filename check only
Yes. I didn't think about cores modifying their "bios" files... not sure what to do in that case.

**Comment by i30817** [comment]
This could be done by really digging into the bioses and seeing which ones have that peculiarity and eliminating that range.

**Comment by orbea** [comment]
I better approach to a bios or content virus would be to harden RetroArch and every core against hostile input.

**Comment by fr500** [comment]
I'll be coming up with a BIOS API call later on.
A lot of the problems with RA are that that the API was pretty much unidirectional, so we have to rely on info files and assumptions.

The idea I have is that there will be a GET_BIOS env callback that will publish the required BIOS files for a system, with the supported CRCs and the lookup locations

Like....

GPGX woud report (this is all made up

```
megadrive.bin, md|gen, CRC1|CRC2, RETRO_SYSTEM_DIR|RETRO_SYSTEM_DIR/Sega Megadrive|RETRO_SYSTEM_DIR/Sega Genesis|RETRO_CONTENT_DIR|RETRO_CONTENT_DIR/Sega Megadrive|RETRO_CONTENT_DIR/Sega Genesis
gamegear.bios, gg, CRC1|CRC2, RETRO_SYSTEM_DIR|RETRO_SYSTEM_DIR/Sega Game Gear|RETRO_SYSTEM_DIR/Sega Game Gear|RETRO_CONTENT_DIR|RETRO_CONTENT_DIR/Sega Game Gear|RETRO_CONTENT_DIR/Sega Game Gear
```

This is the gist of the idea it wouldn't be exactly like that
So when loading content RA would check this callback, would check the content extensions and look for the files in the indicated lookup locations and present error messages accordingly **BEFORE** attempting to load content, would allow to load anyway if the BIOS is optional and would remember the response (hopefully)

**Comment by i30817** [comment]
> A better approach to a bios or content virus would be to harden RetroArch and every core against hostile input.

But bios are executable code, you'd have to run a sandbox against a crafted attack. I agree this is unlikely in this vector though.
